framework/src/problems/FEProblemBase.C

```diff
@@ -2202,35 +2202,22 @@ FEProblemBase::getSampler(const std::string & name, THREAD_ID tid)
 
 bool
 FEProblemBase::duplicateVariableCheck(const std::string & var_name,
-                                      const FEType & type,
                                       bool is_aux)
 {
   SystemBase * curr_sys_ptr = _nl.get();
   SystemBase * other_sys_ptr = _aux.get();
-  std::string error_prefix = "";
   if (is_aux)
   {
     curr_sys_ptr = _aux.get();
     other_sys_ptr = _nl.get();
-    error_prefix = "Aux";
   }
 
   if (other_sys_ptr->hasVariable(var_name))
     mooseError("Cannot have an auxiliary variable and a nonlinear variable with the same name: ",
                var_name);
 
   if (curr_sys_ptr->hasVariable(var_name))
-  {
-    const Variable & var =
-        curr_sys_ptr->system().variable(curr_sys_ptr->system().variable_number(var_name));
-    if (var.type() != type)
-      mooseError(error_prefix,
-                 "Variable with name '",
-                 var_name,
-                 "' already exists but is of a differing type!");
-
-    return true;
-  }
+    mooseError("Cannot have more than one variable with the same name: ", var_name);
```

Comment:
I'd replace this check with a subdomain check. Error out here if the set of subdomains isn't exactly the same.

```
if (curr_sys_ptr->hasVariable(var_name) && current_sys_ptr->get_the_variable(var_name).get_variable_group().active_subdomains() != this_variables_subdomains)
{
  mooseError("BOO!");
}
```

(pseudo code - and you'd have tp pass in the variable's active subdomains)